src/chipper/actions.cljs

```diff
@@ -0,0 +1,173 @@
+;; TODO it would feel more natural to pass state as the first argument to each
+;; of these fns rather than the last
+(ns chipper.actions
+  (:require [clojure.string :refer [split-lines]]
+            [chipper.constants :as const]
+            [chipper.chips :as c]
+            [chipper.keyboard2 :as k]
+            [chipper.state :as s]
+            [chipper.utils :as u]))
+
+;; primitive editor state operations
+;; prefer using these over swap!-ing the state atom directly
+(defn set-mode!
```

Comment:
can put this in `state.cljs` after #32 gets merged. we might want to consider using a lens-like library to get out of this state-munging hell